packages/eas-cli/src/commands/build/index.ts

```diff
@@ -115,7 +128,51 @@ export default class Build extends EasCommand {
       Errors.error('--json is allowed only when building in non-interactive mode', { exit: 1 });
     }
 
+    const validPlatformToBuildResourceClassMapping = new Map(
+      Object.entries({
+        [RequestedPlatform.Android]: new Set([
+          BuildResourceClass.ANDROID_DEFAULT,
+          BuildResourceClass.ANDROID_LARGE,
+        ]),
+        [RequestedPlatform.Ios]: new Set([
+          BuildResourceClass.IOS_DEFAULT,
+          BuildResourceClass.IOS_LARGE,
+          BuildResourceClass.IOS_M1,
+        ]),
+        [RequestedPlatform.All]: new Set(),
```

Comment:
m1 is obviously only for ios, but there is no reason to not support default/large for all option

Reply:
It seems like a nice-to-have to me, at least for right now. The same functionality is achieved by just running two `eas build` commands, one for android and one for iOS. I'm curious if @brentvatne has other thoughts on this.